### DISCUSSION TITLE
컨트롤러는 클러스터 안에 작동하는 브로커 중 하나가 자동으로 선정된다고 하는데 어떻게 선정되는 것인가?

### DISCUSSION THREAD

**Discussion by bonyeon0605** [kyungjunleeme/kafka_study#5] [Q&A]
자동 선출 방식이란?
클러스터에서 가장 먼저 주키퍼의 /controller에 노드를 생성한 브로커가 컨트롤러 역할을 맡게 된다.
주키퍼 내에 와치를 설정하여 컨트롤러 브로커에 변동이 발생했을 때 다른 브로커에게 알리는 기능도 제공

관련 블로그 
https://devidea.tistory.com/71

**Comment by kyungjunleeme** [comment]
![image](https://github.com/user-attachments/assets/f43e4a29-c9ae-42c3-933c-83fb2efc8370)
- elect()

- ![image](https://github.com/user-attachments/assets/c2e80c9f-1040-47f5-be2f-fc78a8c9ad0a)
- 이것도 다시 한번 정리, 준영님 말씀대로 group coordianotr 로서(as - as는 역할, 지위를 나태는 전치사) broker에게 heart beat를 보내, membership 을 유지한다.(영어 공식문서 읽을때 자주 쓰이는 membership check - ex. check membership of a set in python )
